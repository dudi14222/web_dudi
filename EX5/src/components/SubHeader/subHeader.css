.sub-header {
    background-image: url(/images/adrenaline-renegade-wallpaper-choppers.jpg);
    height: 270px;
    background-size: cover;
    background-position: center;
    color: #fafafa;
    display: flex;
    justify-content: center;
    align-items: center;
}
