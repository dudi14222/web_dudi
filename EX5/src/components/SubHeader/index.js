import React from 'react';
import './subHeader.css';
const SubHeader = ({children}) => (
    <div className="sub-header"> 
        <h1>{children}</h1>        
    </div>
);

export default SubHeader;
