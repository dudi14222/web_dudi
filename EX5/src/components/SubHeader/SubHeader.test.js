import SubHeader from '../SubHeader';
import React from 'react';
import { MemoryRouter } from 'react-router-dom'
import renderer from 'react-test-renderer';
import { shallow } from "enzyme";



describe('SubHeader snapshot tests', () => {
    it('test snapshot', () => {
        const tree = renderer.create(
            <MemoryRouter>
                <SubHeader />
            </MemoryRouter>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    })
})

describe('SubHeader shallow tests', () => {
    it('should render children when passed in', () => {
        const wrapper = shallow((
            <SubHeader>
                <div className="unique" />
            </SubHeader>
        ));    
        expect(wrapper.contains(<div className="unique" />)).toBe(true);
    });
})



