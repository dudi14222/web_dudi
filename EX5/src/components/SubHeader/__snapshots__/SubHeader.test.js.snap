// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubHeader snapshot tests test snapshot 1`] = `
<div
  className="sub-header"
>
  <h1 />
</div>
`;
